### PR TITLE
Watcher: Move PipelineRun watching to v1alpha2 API.

### LINKED ISSUE #23: [v1alpha2] Migrate watcher to v1alpha2 API
Tracking bug for migrating the watcher from the v1alpha1 Results API to v1alpha2.

### PR DESCRIPTION
**Pull request by wlynch**
This largely follows what we previously did with TaskRuns. This does not
handle linking TaskRuns with PipelineRuns in the same Result - this will
be done in another PR.

While this removes the last usages of the v1alpha1 API, we'll remove the
API/impl in another PR.

Fixes #23

### CODE REVIEW

File: pkg/watcher/reconciler/pipelinerun/pipelinerun.go [hunk 1]
```diff
@@ -30,85 +29,53 @@ import (
...
 		return nil
 	}
-	logger.With(zap.String("Namespace", namespace), zap.String("Name", name))
 
-	pr, err := r.pipelineclientset.TektonV1beta1().PipelineRuns(namespace).Get(name, metav1.GetOptions{})
+	// Lookup current PipelineRun.
+	tr, err := r.pipelineclientset.TektonV1beta1().PipelineRuns(namespace).Get(name, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		logger.Errorf("PipelineRun in work queue no longer exists: %v", err)
+		log.Warnf("TaskRun not found: %v", err)
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun.go:49]: It should be PipelineRun?

File: pkg/watcher/reconciler/pipelinerun/pipelinerun.go [hunk 2]
```diff
@@ -30,85 +29,53 @@ import (
...
 		return nil
 	}
 	if err != nil {
-		logger.Errorf("Error retrieving PipelineRun: %v", err)
+		log.Errorf("TaskRun.Get: %v", err)
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun.go:53]: PipelineRun?

File: pkg/watcher/reconciler/pipelinerun/pipelinerun.go [hunk 3]
```diff
@@ -30,85 +29,53 @@ import (
...
 	if err != nil {
-		logger.Errorf("Invalid resource key: %s", key)
+		log.Errorf("invalid resource key: %s", key)
 		return nil
 	}
-	logger.With(zap.String("Namespace", namespace), zap.String("Name", name))
 
-	pr, err := r.pipelineclientset.TektonV1beta1().PipelineRuns(namespace).Get(name, metav1.GetOptions{})
+	// Lookup current PipelineRun.
+	tr, err := r.pipelineclientset.TektonV1beta1().PipelineRuns(namespace).Get(name, metav1.GetOptions{})
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun.go:47]: Better be pr?

File: pkg/watcher/reconciler/pipelinerun/pipelinerun.go [hunk 4]
```diff
@@ -30,85 +29,53 @@ import (
...
-			return err
-		}
-		logger.Infof("Creating a new result: %s", prResult.GetName())
+	if a := tr.GetAnnotations(); result.GetName() == a[annotation.Result] && record.GetName() == a[annotation.Record] {
+		// Result annotations are already present in the TaskRun.
+		// Nothing else to do.
+		return nil
 	}
 
+	// Update TaskRun with Result Annotations.
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun.go:70]: Same here

File: pkg/watcher/reconciler/pipelinerun/pipelinerun.go [hunk 5]
```diff
@@ -30,85 +29,53 @@ import (
...
-			logger.Errorf("Error jsonpatch for PipelineRun Result %s: %v", prResult.GetName(), err)
-			return err
-		}
-		if _, err := r.pipelineclientset.TektonV1beta1().PipelineRuns(pr.Namespace).Patch(pr.Name, types.JSONPatchType, path); err != nil {
-			logger.Errorf("Error apply the patch to PipelineRun: %v", err)
-			return err
-		}
-		logger.Infof("Creating a new result: %s", prResult.GetName())
+	if a := tr.GetAnnotations(); result.GetName() == a[annotation.Result] && record.GetName() == a[annotation.Record] {
+		// Result annotations are already present in the TaskRun.
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun.go:65]: Same here

File: pkg/watcher/reconciler/pipelinerun/pipelinerun.go [hunk 6]
```diff
@@ -30,85 +29,53 @@ import (
...
 	}
 
+	// Update TaskRun with Result Annotations.
+	patch, err := annotation.Add(result.GetName(), record.GetName())
+	if err != nil {
+		log.Errorf("error adding Result annotations: %v", err)
+		return err
+	}
+	if _, err := r.pipelineclientset.TektonV1beta1().PipelineRuns(tr.GetNamespace()).Patch(tr.Name, types.JSONPatchType, patch); err != nil {
+		log.Errorf("TaskRun.Patch: %v", err)
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun.go:77]: same here

File: pkg/watcher/reconciler/pipelinerun/pipelinerun_test.go [hunk 7]
```diff
@@ -20,116 +20,144 @@ import (
...
 }
 
-func TestReconcile_UnchangePipelineRun(t *testing.T) {
-	tt := newPipelineRunTest(t)
+func TestReconcile(t *testing.T) {
+	env := newEnv(t)
 
-	// Reconcile once to get IDs, etc.
-	pr, err := test.ReconcilePipelineRun(tt.ctx, tt.asset, tt.pipelineRun)
+	tr, err := env.pipeline.TektonV1beta1().PipelineRuns("ns").Create(&v1beta1.PipelineRun{
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun_test.go:70]: pr here?

File: pkg/watcher/reconciler/pipelinerun/pipelinerun_test.go [hunk 8]
```diff
@@ -20,116 +20,144 @@ import (
...
-	newpr, err := test.ReconcilePipelineRun(tt.ctx, tt.asset, tt.pipelineRun)
-	if err != nil {
-		t.Fatalf("failed to get second PipelineRun %s: %v", tt.pipelineRun.Name, err)
-	}
-	if diff := cmp.Diff(pr, newpr); diff != "" {
-		t.Fatal(diff)
-	}
-}
+	t.Run("create", func(t *testing.T) {
+		tr = reconcile(t, env, tr)
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun_test.go:112]: pr :) ?

File: pkg/watcher/reconciler/pipelinerun/pipelinerun_test.go [hunk 9]
```diff
@@ -20,116 +20,144 @@ import (
...
+// reconcile forces a reconcile for the given PipelineRun, and returns the newest
+// PipelineRun post-reconcile.
+func reconcile(t *testing.T, env *env, want *v1beta1.PipelineRun) *v1beta1.PipelineRun {
+	if err := env.ctrl.Reconciler.Reconcile(env.ctx, want.GetNamespacedName().String()); err != nil {
+		t.Fatalf("Reconcile: %v", err)
 	}
-	updatepr, err := test.ReconcilePipelineRun(tt.ctx, tt.asset, pr)
+
+	// Verify that the PipelineRun now has a Result annotation associated with it.
+	tr, err := env.pipeline.TektonV1beta1().PipelineRuns(want.GetNamespace()).Get(want.GetName(), metav1.GetOptions{})
```
Comment (XinruZhang) [inline pkg/watcher/reconciler/pipelinerun/pipelinerun_test.go:137]: pr?

**Review by XinruZhang — COMMENTED** [review]
Looks good, just several typos

File: pkg/watcher/reconciler/pipelinerun/pipelinerun.go (hunk 1 above)
Comment (wlynch) [reply pkg/watcher/reconciler/pipelinerun/pipelinerun.go:49]: I did catch these but pushed them to a wrong branch. 🙃 
 Fixed.

**Review by XinruZhang — APPROVED** [review]
/lgtm

**Pull request merged**